REMOVE & PUT calls to /api/restaurants now remove related suggestions

otsha opened the pull request:
- New helper methods 
  -`tryRemoveSuggestionsByRestaurant` so that all suggestions related to a restaurant that's being deleted or updates are removed altogether
  -`tryUpdateRestaurant` to be used in the future update suggestion feature

otsha committed:
REMOVE & PUT calls to /api/restaurants now removes related suggestions

ritinae commented on packages/backend/src/controllers/restaurants.js:
```diff
@@ -38,6 +45,16 @@ const removeRestaurantFromCategories = async (restaurantId, categoryIds) => {
+
+  if (!removedSuggestions) {
+    return null
+  }
```
```suggestion
```
delet tis

ritinae reviewed (changes requested):
Slight code smell in controller

ritinae committed:
Remove dead code

`removedSuggestions` is never undefined, thus returning null does not differ from the return a few lines below
